pkg/codegen/go/gen.go

```diff
@@ -4513,7 +4535,10 @@ func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error
 			pkg.getImports(resource, importsAndAliases)
 			importsAndAliases["github.com/pulumi/pulumi/sdk/v3/go/pulumi"] = ""
 			importsAndAliases[path.Join(pkg.importBasePath, pkg.internalModuleName)] = ""
-			importsAndAliases["github.com/pulumi/pulumi/sdk/v3/go/pulumix"] = ""
+			if goPkgInfo.Generics == GenericsSettingSideBySide {
```

Comment:
Isn't this also needed for GenericsOnly?

Reply:
No because this is `genResource` when `useGenericVariant` is set to false. The code underneath continues with adding the `pulumix` import when `useGenericVariant` is set to true